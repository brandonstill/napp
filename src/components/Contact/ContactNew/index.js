import React, { Component } from 'react';
import { connect } from 'react-redux';
import { saveContact } from '../../../store/actions';

import { withStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';

const styles = theme => ({
  textField: {
    display: 'flex',
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
    width: '100%',
  },
  actionItems: {
    marginTop: '20px',
    textAlign: 'right'
  },
  cancelButton: {
    margin: theme.spacing.unit,
  },
});

export class ContactNew extends Component {
  state = {
    firstName: '',
    lastName: '',
    email: ''
  }

  handleChange = (name, event) => event => {
    console.log(name);
    console.log(event);
    this.setState({
      [name]: event.target.value,
    });
  };

  handleSubmit = event => {
    event.preventDefault();
    this.props.saveContact(this.state);
  }

  renderField = (label, type, name) => {

    return(
      // <div>
      //   <label htmlFor={name}>{label}</label>
      //   <input value={this.state[name]} type={type} name={name} id={name} onChange={this.handleChange} />
      // </div>
      
      <TextField
        required
        id={name}
        name={name}
        label={label}
        className={this.props.classes.textField}
        value={this.state.name}
        onChange={this.handleChange(name)}
        margin="normal"
        style={{
          marginLeft: 0,
          marginRight: 0
        }}
      />

    )
  }

  render() {
    return( 
      // <form onSubmit={this.handleSubmit}>
      //   {this.renderField('First Name', 'text', 'firstName')}
      //   {this.renderField('Last Name', 'text', 'lastName')}
      //   {this.renderField('Email', 'text', 'email')}
      //   <div>
      //     <button type="submit">Submit</button>
      //     <button type="button" onClick={this.props.handleClose}>Cancel</button>
      //   </div>
      // </form>
      <form className={this.props.classes.container} noValidate autoComplete="off" onSubmit={this.handleSubmit}>
        <Typography gutterBottom variant="headline" component="h2">
          Add New Contact
        </Typography>
        {this.renderField('First Name', 'text', 'firstName')}
        {this.renderField('Last Name', 'text', 'lastName')}
        {this.renderField('Email', 'text', 'email')}
        <div className={this.props.classes.actionItems}>
          <Button 
            size="large" 
            className={this.props.classes.cancelButton} 
            onClick={this.props.handleClose} 
            type="button"
            name="cancel"
          >
            Cancel
          </Button>
          <Button 
            variant="contained" 
            size="large" 
            color="primary" 
            type="submit"
            name="submit"
          >
           Add
          </Button>
        </div>
      </form>
    )
  }
}

const mapDispatchToProps = dispatch => {
  return {
    saveContact: (data) => dispatch(saveContact(data))
  }
}

const styledComponent = withStyles(styles)(ContactNew);
export default connect(null, mapDispatchToProps)(styledComponent);