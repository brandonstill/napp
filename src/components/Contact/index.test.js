import React from 'react';
import { shallow } from 'enzyme';
import { Contact } from './index.js';
import ContactNew from './ContactNew';
import { Paper } from '@material-ui/core';

describe('Contacts', () => {

  it('renders correctly', () => {
    const wrapper = shallow(<Contact fetchContacts={() => {}} classes={{paper:''}} />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it('should call fetchContacts action', () => {
    const spy = jest.fn();
    const wrapper = shallow(<Contact fetchContacts={spy} classes={{paper:''}} />);
    expect(spy).toHaveBeenCalled();
  });

  it('to have correct sortBy state', () => {
    const wrapper = shallow(<Contact fetchContacts={() => {}} classes={{paper:''}} />).instance();
    expect(wrapper.state.sortBy).toEqual('lastName');
  });

  it('handleSort updates component state', () => {
    const wrapper = shallow(<Contact fetchContacts={() => {}} classes={{paper:''}} />).instance();
    wrapper.handleSort({ target: { value: 'firstName' } });
    expect(wrapper.state.sortBy).toEqual('firstName');
  });

  it('handleOpen calls contactModal action with true', () => {
    const spy = jest.fn();
    const wrapper = shallow(<Contact fetchContacts={() => {}} contactModal={spy} classes={{paper:''}} />).instance();
    wrapper.handleOpen();
    expect(spy).toHaveBeenCalledWith(true);
  });

  it('handleClose calls contactModal action with false', () => {
    const spy = jest.fn();
    const wrapper = shallow(<Contact fetchContacts={() => {}} contactModal={spy} classes={{paper:''}} />).instance();
    wrapper.handleClose();
    expect(spy).toHaveBeenCalledWith(false);
  });

});