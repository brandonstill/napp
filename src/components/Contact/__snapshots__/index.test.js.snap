// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts renders correctly 1`] = `
Array [
  <WithStyles(Grid)
    container={true}
    spacing={24}
    style={
        Object {
            "margin": "0px",
            "padding": "20px",
            "width": "100%",
          }
    }
>
    <WithStyles(ContactNav)
        handleOpen={[Function]}
        handleSort={[Function]}
        sortBy="lastName"
    />
    <WithStyles(ContactGrid)
        contacts={undefined}
        sortBy="lastName"
    />
    <WithStyles(Modal)
        aria-describedby="new-contact-modal"
        aria-labelledby="new-contact-modal"
        onClose={[Function]}
        open={undefined}
    >
        <div
            className=""
        >
            <WithStyles(ContactNew)
                handleClose={[Function]}
                handleSave={[Function]}
            />
        </div>
    </WithStyles(Modal)>
</WithStyles(Grid)>,
]
`;
